dove/tests/cmd_dove_clean.rs

```diff
@@ -0,0 +1,110 @@
+#![cfg(test)]
+
+use std::path::{Path, PathBuf};
+use std::fs::{remove_dir_all, read_to_string};
+use fs_extra::file::write_all;
+use dove::cli::execute;
+
+/// $ cargo run -- clean
+/// $ dove clean
+/// project name: demoproject_16
+#[test]
+fn clean() {
+    // Path to dove folder
+    let dove_folder = {
+        let mut folder = Path::new(".").canonicalize().unwrap();
+        if folder.to_str().unwrap().find("dove").is_none() {
+            folder.push("dove");
+        }
+        folder
+    };
+    // Project name and path
+    let project_name = "demoproject_16";
+    let project_folder = {
+        let mut folder = dove_folder.clone();
+        folder.push(project_name);
+        if folder.exists() {
+            remove_dir_all(&folder).expect(&format!(
+                "[ERROR] Couldn't delete project directory: {}",
+                folder.to_str().unwrap()
+            ));
+        }
+        folder
+    };
+    // $ cargo run -- new demoproject_16 -d pont
+    // $ dove new demoproject_16 -d pont
+    {
+        let args = &["dove", "new", project_name];
+        let command_string: String = args.join(" ").to_string();
+        execute(args, dove_folder.clone()).expect(&format!("[COMMAND] {}", &command_string));
```

Comment:
It's not related to this PR, but why `execute` accepts `String` as the first argument, while you have to use `args.join(" ").to_string();` everywhere?

Reply:
It accepts `Vec<String>` and `command_string` is only created for the error message.

Reply:
Although, It would be great to move it inside the `format!` call to remove the variable that is only used in an error message.

Reply:
I see, thank you. Though slices can be rendered using `Debug` implementation (`{:?}`), so there is to need to join them anyway. We will only see this string in case of error, and I think we can afford a slight readability loss (`dove new project` vs `["dove", "new", "project"]`)